File: frontend/projects/storage/src/lib/entities/storage-watcher-event.ts
import {StorageNode} from 'projects/storage/src/lib/entities/storage-node';
import {Owner} from 'projects/security/src/lib/entities/owner';
import {StorageWatcherEventType} from 'projects/storage/src/lib/entities/storage-watcher-event-type';

export interface StorageWatcherEvent {
  node: StorageNode;
  type: StorageWatcherEventType;
  owner: Owner;
}

File: backend/commons/storage/storage-entity/src/main/java/com/octoperf/kraken/storage/entity/StorageWatcherEvent.java
package com.octoperf.kraken.storage.entity;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.octoperf.kraken.security.entity.owner.Owner;
import com.octoperf.kraken.tools.event.bus.BusEvent;
import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

@Value
@Builder(toBuilder = true)
public class StorageWatcherEvent implements BusEvent {

  StorageNode node;
  StorageWatcherEventType type;
  Owner owner;

  @JsonCreator
  StorageWatcherEvent(
      @NonNull @JsonProperty("node") final StorageNode node,
      @NonNull @JsonProperty("type") final StorageWatcherEventType type,
      @NonNull @JsonProperty("owner") final Owner owner
  ) {
    super();
    this.node = node;
    this.type = type;
    this.owner = owner;
  }

}
